### PR TITLE
Add decorators & Link them to classes

### PR DESCRIPTION
**Pull request by Peter1907**
Hello, dear code reviewer.
In this branch, I've implemented the following

- Added a "Nameable" class & linked it to the "Person" class
- Added the main "Decorator"
- Added a "CapitalizeDecorator" that capitalizes the first letter of the name
- Added a "TrimmerDecorator" that returns the name with a maximum of 10 letters

Let me know if anything is to be modified.
Thank you!

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def correct_name
+    @namable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
```
Comment (TedLivist) [inline decorator.rb:18]: - [ ] **[OPTIONAL]** Based on Ruby conventions classes should be in their own files. I highly recommend you move this class to its own file.

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if super.length > 10
+      super[0..9]
+    else
+      super
+    end
+  end
```
Comment (TedLivist) [inline decorator.rb:27]: - [ ] **[OPTIONAL]** Based on Ruby conventions classes should be in their own files. I highly recommend you move this class to its own file.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Peter1907,

✔️ Implementation of class decorator

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Peter1907** [comment]
Okay thank you @TedLivist for the suggestions I'll implement them

**Pull request merged**